Objective-C/CBLQueryResultSet.mm

```diff
@@ -87,17 +92,22 @@ - (CBLDatabase*) database {
 
 
 - (id) nextObject {
-    if (_randomAccess)
+    if (self.randomAccess)
         return nil;
     
-    id row = nil;
-    if (c4queryenum_next(_c4enum, &_error)) {
-        row = self.currentObject;
-    } else if (_error.code)
-        CBLWarnError(Query, @"%@[%p] error: %d/%d", [self class], self, _error.domain, _error.code);
-    else
-        CBLLog(Query, @"End of query enumeration (%p)", _c4enum);
-    return row;
+    CBLDatabase* strongDB = self.database;
+    Assert(strongDB, @"Database has been released.");
```

Comment:
The message seems wrong -- it's really that the query's been dealloced, not the db.
And is it required that the app hold onto the CBLQuery object for as long as it's using the result set, or else it'll crash? That seems kind of fragile.